arch/x86/pagetables.c

```diff
@@ -159,8 +160,8 @@ static mfn_t get_pgentry_mfn(mfn_t tab_mfn, pt_index_t index, unsigned long flag
         frame_t *frame = get_free_frame();
         BUG_ON(!frame);
 
-        set_pgentry(entry, frame->mfn, flags);
-        mfn = mfn_from_pgentry(*entry);
+        mfn = frame->mfn;
+        set_pgentry(entry, mfn, flags);
```

Comment:
Unrelated change?

Reply:
It's just a code simplification. No functional change (intended).